[O11y][Oracle] Add custom number formatting to visualisations

kush-elastic opened the pull request:
## Description
### Workaround implemented during lens migration
Used default format and migrated visualization to the lens.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).


## Updated number formatting for the following dashboards

- [Metrics Oracle] Performance Metrics
- [Metrics Oracle] System Statistics Metrics
- [Metrics Oracle] Sysmetric Metrics
- [Metrics Oracle] Tablespace
- [Metrics Oracle] Overview

## Related issues

<!-- Recommended
Link related issues below. Insert the issue link or reference after the word "Closes" if merging this should automatically close it.

- Closes #123
- Relates #123
- Requires #123
- Supersedes #123
-->
- Closes #7117 

## Note

According to this [comment](https://github.com/elastic/integrations/pull/5830#pullrequestreview-1378314209) received for MongoDB, we also checked for the panels in Oracle that require custom number formatting. However, it had an open [issue](https://github.com/elastic/kibana/issues/72711) which was released in the 8.10 version of Kibana. Hence updated the Kibana version to 8.10.2 for the change to be available in Kibana.

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8381/detail/PR-8381/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8381/detail/PR-8381/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8381/detail/PR-8381/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://fleet-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations%2FPR-8381/detail/PR-8381/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_8381.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/fleet-ci/transactions/view?rangeFrom=2023-11-02T10:07:04.100Z&rangeTo=2023-11-02T10:27:04.100Z&transactionName=BUILD+Ingest-manager%2Fintegrations%2FPR-%7Bnumber%7D&transactionType=job&latencyAggregationType=avg&traceId=b0d12697fd78e7cfa4bbd3f77c53411b&transactionId=9b69f9d8265f77a6)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-11-02T10:17:04.100+0000


* Duration: 37 min 14 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 26  |
| Skipped      | 0 |
| Total        | 26   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->














## :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.




</p>
</details>

<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
### :globe_with_meridians: Coverage report

Name | Metrics % (`covered/total`) | Diff
 --- | --- | ---
Packages | 100.0% (`1/1`) | :green_heart: 
Files | 100.0% (`1/1`) | :green_heart: 
Classes | 100.0% (`1/1`) | :green_heart: 
Methods | 100.0% (`29/29`) | :green_heart: 2.632
Lines | 94.424% (`254/269`) | :-1: -1.003
Conditionals | 100.0% (`0/0`) | :green_heart: 

<!--COMMENT_GENERATED_WITH_ID_coverage-->

milan-elastic reviewed (approved):
LGTM!

agithomas reviewed (commented):
Parse time value (bar chart) is showing negative values. Similarly others as well. Please revisit

agithomas reviewed (commented):
Sysmetric dashboard is not consistant in terms of usage of theme colors. Please align the color themes

agithomas reviewed (commented):
Ratio of used space dashboard is not showing any value. Please revist

agithomas reviewed (commented):
Please align the Lock/Pin dashboard of Oracle Overview dashboard seems to be not aligned with the colors. Please revisit

kush-elastic commented:
> Parse time value (bar chart) is showing negative values. Similarly others as well. Please revisit

Please refer to the previous [conversation](https://github.com/elastic/integrations/pull/5815#issuecomment-1506444750) for the same. This is applicable to all the visualizations that are using differences currently.

> Sysmetric dashboard is not consistant in terms of usage of theme colors. Please align the color themes
> Ratio of used space dashboard is not showing any value. Please revist
> Please align the Lock/Pin dashboard of Oracle Overview dashboard seems to be not aligned with the colors. Please revisit

Updated.

agithomas commented:
Requesting for a revalidation. Please refer to the panel below. All the values are exactly the same to the 1/1000th place.  Requesting for a re-check if the viz formula are indeed correct.  Please confirm

<img width="412" alt="image" src="https://github.com/elastic/integrations/assets/101976829/3f11e30a-a97a-4e27-a308-9ea0a3ba32d1">

agithomas commented:
For other panels when Y axis value is percentage, `%` is added in the Y-axis values. Please add it here to ensure consistency.

agithomas commented:
Please refer here . Is the Y-axis value '%' ? if Yes, please add that . 

<img width="671" alt="image" src="https://github.com/elastic/integrations/assets/101976829/bc735627-d15e-40d4-9454-fbdaa30274d2">

kush-elastic commented:
> [Requesting for a revalidation. Please refer to the panel below. All the values are exactly the same to the 1/1000th place. Requesting for a re-check if the viz formula are indeed correct. Please confirm](https://github.com/elastic/integrations/pull/8381#issuecomment-1825236101)

Checked the formula and it is as expected. I have updated the screenshot by populating the data so it shows different values and sorts them properly. Updated the visualization in Overview and Tablespace dashboard as it is present at both places. Also one more thing I noticed, as per this [comment](https://github.com/elastic/integrations/pull/8381#pullrequestreview-1731734279), the visualization was not showing values when it reaches the top limit of 10 values. Hence, increased the size of visualization as well to accumulate the values.

> [For other panels when Y axis value is percentage, % is added in the Y-axis values. Please add it here to ensure consistency.](https://github.com/elastic/integrations/pull/8381#issuecomment-1825237743)

Added the same

> [Please refer here . Is the Y-axis value '%' ? if Yes, please add that.](https://github.com/elastic/integrations/pull/8381#issuecomment-1825240894)

For this visualization, please refer to the documentation [here](https://docs.oracle.com/cd/B16240_01/doc/doc.102/e16282/oracle_database_help/oracle_database_instance_efficiency_cpuusage_ps.html#:~:text=Description,being%20performed%20by%20the%20database.). I don’t find it as a percentage field.


Also while looking at the Ratio of used space in data file visualization Tablespace dashboard, I found that the Rank By is not set to use proper fields, hence updated that as well. Please have a look at the settings before and after.

Before:-
![image](https://github.com/elastic/integrations/assets/89848966/365bdc5b-95c0-48a6-b62a-b3a6519ffbbf)

After:-
![image](https://github.com/elastic/integrations/assets/89848966/5b44fedb-8022-48d3-bbbc-53f50d2f2808)

kush-elastic commented on packages/oracle/manifest.yml:
```diff
@@ -1,15 +1,15 @@
 conditions:
   kibana:
-    version: "^8.8.0"
+    version: "^8.10.2"
```
Right, I have updated the PR description with a reason.

agithomas reviewed (approved):
LGTM!

elasticmachine commented:
Package oracle - 1.23.0 containing this change is available at https://epr.elastic.co/search?package=oracle
